src/TMS/CourseCreation/CourseAccessExtension.php

```diff
@@ -4,18 +4,54 @@
 
 namespace ILIAS\TMS\CourseCreation;
 
+require_once("Services/Component/classes/class.ilPluginAdmin.php");
+
 /**
  * Enhances a course access handler with methods required for the course creation.
  */
 trait CourseAccessExtension {
-	function _checkAccessExtension($a_cmd, $a_permission, $a_ref_id, $a_obj_id, $a_user_id) {
-		// TODO: implement me properly
-		// Check if CourseCreation-plugin exists
-		// Check if course is a template
-		// Check if user may copy course
-		// Check if user can see course creation plugin
+	static function _checkAccessExtension($a_cmd, $a_permission, $a_ref_id, $a_obj_id, $a_user_id) {
```

Comment:
sprinkle with some asserts, please.

Reply:
This won't do any good IMO. Best I could do here is assert for strings, as this is called from ILIAS. I pushed the assertions down one level in the call stack.